linode/manager
# M3-5839: Add More Insightful Credit Card Expiration Date Validation

## Description 📝

- The goal of this PR is to decrease friction when inputting a Credit Card's expiration date
  - Support more formats
  - Allow user to submit form at any time (don't disable button by default)
  - Add validation message when there is no delimiter (`/`)

### The Issue ❗️
- Users were confused when they entered something like `052025` as a credit card expiration date and the `Add Credit Card` button was still disabled

## The Fix 🔧
- Allow users to push the `Add Credit Card` button with a partial expiry inputed so they can get some validation feedback
- If the user does not include some kind of delimiter to separate the month and year, show a validation message (`Date format must have forward slash between the month and year.`)

## Additional Feature ✨
- Users can now user characters other than just a `/` to seperate the month and year
  - The following are now acceptable values for the expiry date: `5 25`, `5-25`, `05 2025`, `05 25`, ...

## Preview 📷

https://user-images.githubusercontent.com/6440455/183980648-81b6fa72-8646-407b-b054-7113c810c063.mov

## How to test 🧪

- **Ensure you can still add a Credit Card to your Linode account**
- Check overall validation of the credit card form
- Ensure when  a user enters something like `052026`, that they are presented with a descriptive error message telling them how to resolve the formatting issue
